src/core/thread/lowpan.cpp

```diff
@@ -613,7 +613,7 @@ int Lowpan::DecompressBaseHeader(Ip6::Header &       aIp6Header,
     uint8_t *            bytes;
 
     VerifyOrExit(remaining >= 2);
-    hcCtl = static_cast<uint16_t>((cur[0] << 8) | cur[1]);
+    hcCtl = Encoding::BigEndian::ReadUint16(cur);
```

Comment:
nit: we have `ot::Encoding::BigEndian::HostSwap16` at the top.  Do the same for `ReadUint16` to be consistent?